sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2834,11 +2834,11 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
 
       case Some(query) =>
         ReplaceTableAsSelectStatement(table, query, partitioning, bucketSpec, properties,
-          provider, options, location, comment, orCreate = orCreate)
+          Some(provider), options, location, comment, orCreate = orCreate)
```

Comment:
These should be `Option` not `Some`

Reply:
Makes sense. Thanks! Though I think currently it wouldn't be null as I can't even do a replace table without the `USING` clause.